### PR TITLE
Fix lineage for UniqueExpressionTermsVisitor

### PR DESCRIPTION
**Pull request by lbschanno**
Fix UniqueExpressionTermsVisitor to ensure that it returns a JEXL tree
with a valid lineage, and does not modify the original input tree.

Part of #880 and #968

### CODE REVIEW

File: warehouse/query-core/src/test/java/datawave/query/jexl/visitors/UniqueExpressionTermsVisitorTest.java [hunk 1]
```diff
@@ -236,21 +239,31 @@ public void testDuplicateSubTreeDisjunction() throws ParseException {
...
-        
-        // Assert that both original and expected query strings parse as expected
-        assertEquals("Original query did not parse cleanly" + original, original, JexlStringBuildingVisitor.buildQuery(originalScript));
-        assertEquals("Expected query did not parse cleanly" + expected, expected, JexlStringBuildingVisitor.buildQuery(expectedScript));
         
         // Remove duplicate terms from within expressions.
         ASTJexlScript visitedScript = UniqueExpressionTermsVisitor.enforce(originalScript);
         
-        // De-duped query should match expected
-        assertEquals(expected, JexlStringBuildingVisitor.buildQuery(visitedScript));
```
Comment (apmoriarty) [inline warehouse/query-core/src/test/java/datawave/query/jexl/visitors/UniqueExpressionTermsVisitorTest.java:253]: This check should be left in. There have been cases were the tree equality visitor said trees were equal but the queries parsed into different strings.

**Review by apmoriarty — CHANGES_REQUESTED** [review]
Good refactor, I would like to see an assertion added back to the test.

File: warehouse/query-core/src/test/java/datawave/query/jexl/visitors/UniqueExpressionTermsVisitorTest.java (hunk 1 above)
Comment (lbschanno) [reply warehouse/query-core/src/test/java/datawave/query/jexl/visitors/UniqueExpressionTermsVisitorTest.java:253]: No problem, it's been added back in.

**Review by ivakegg — APPROVED** [review]

**Review by apmoriarty — APPROVED** [review]
LGTM

**Pull request merged**